### PR TITLE
Upgrade CI to ROCm5.3

### PR DESCRIPTION
**Pull request by jithunnair-amd**
cc @jeffdaily @sunway513 @pruthvistony @ROCmSupport

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@seemethere @kit1980 Do we need to create docker repository for ROCm5.3 + py3.8? I'm seeing this in the logs:
```
2022-11-02T07:40:36.0513827Z + docker manifest inspect 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm5.3-py3.8:8d1c64832dba096918d578884b2c3775f561fc0a
2022-11-02T07:40:36.2910816Z + docker push 308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm5.3-py3.8:8d1c64832dba096918d578884b2c3775f561fc0a
2022-11-02T07:40:36.3110046Z The push refers to repository [308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm5.3-py3.8]
```

Maybe we should bulk-create for multiple ROCm releases like we did in the past? I believe the move from py3.7 to py3.8 recently might have necessitated this.

**Comment by malfet** [comment]
Created `pytorch-linux-focal-rocm5.3-py3.8` ECR, but I wonder if we can change the naming schema to `pytorch-linux-focal-rocm5-py3` to reduce number of times we need to make changes

**Comment by jithunnair-amd** [comment]
Docker build jobs succeeded! Now running UTs

**Comment by jithunnair-amd** [comment]
ROCm CI unit test failures:
[linux-focal-rocm5.3-py3.8 / test (default, 1, 2, linux.rocm.gpu)](https://github.com/pytorch/pytorch/actions/runs/3446672395/jobs/5752744800):
test_comprehensive_pow_cuda_complex32 (__main__.TestDecompCUDA)
test_quick_pow_cuda_complex32 (__main__.TestDecompCUDA)

[linux-focal-rocm5.3-py3.8 / test (default, 2, 2, linux.rocm.gpu)](https://github.com/pytorch/pytorch/actions/runs/3446672395/jobs/5752744966)
test_linalg_solve_triangular_cuda_float32 (__main__.TestLinalgCUDA)

[linux-focal-rocm5.3-py3.8-distributed / test (distributed, 1, 2, linux.rocm.gpu)](https://github.com/pytorch/pytorch/actions/runs/3446672364/jobs/5752746528)
test_diff_hyperparams_cpu_offload_sharding_strategy_str_no_shard (__main__.TestFSDPUseOrigParamsMultipleParamGroups)

[linux-focal-rocm5.3-py3.8-distributed / test (distributed, 2, 2, linux.rocm.gpu)](https://github.com/pytorch/pytorch/actions/runs/3446672364/jobs/5752746682)
test_transformer_offload_true_no_shard_norm_type_None (__main__.TestParityWithDDP)
test_nested_always_wrap_model_offload_true_no_shard_norm_type_None (__main__.TestParityWithDDP)

**Review by malfet — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
> Created `pytorch-linux-focal-rocm5.3-py3.8` ECR, but I wonder if we can change the naming schema to `pytorch-linux-focal-rocm5-py3` to reduce number of times we need to make changes

Filed PR https://github.com/pytorch/pytorch/pull/90694 for the same

**Comment by jithunnair-amd** [comment]
All ROCm CI jobs passed. Merging

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -10222,7 +10222,11 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            sample_inputs_func=sample_inputs_masked_select,
-           error_inputs_func=error_inputs_masked_select),
+           error_inputs_func=error_inputs_masked_select,
+           skips=(
+               # Compiler issue on ROCm. Might need to skip until ROCm5.5
```
Comment (IvanYashchuk) [inline torch/testing/_internal/common_methods_invocations.py:10227]: @jithunnair-amd All of these skips should have been applied only for the ROCm platform. Please look at how other tests are skipped (`active_if=TEST_WITH_ROCM`).

**Review by IvanYashchuk — CHANGES_REQUESTED** [review]
Tests from `torch/testing/_internal/common_methods_invocations.py` should be skipped only for ROCm.

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (jithunnair-amd) [reply torch/testing/_internal/common_methods_invocations.py:10227]: Thanks @IvanYashchuk, my fault. I have added the ROCm-only skip logic.

**Pull request closed without merging**